Sources/Plugins/AttachmentManager/AttachmentManager.swift

```diff
@@ -121,6 +121,17 @@ open class AttachmentManager: NSObject, InputPlugin {
         })
     }
     
+    /// Performs an animated removal of all attachments
+    open func removeAllAttachments() {
+        let attachments = self.attachments
```

Comment:
It won't be `nil` as I make a copy of the attachment array first.
I do this in this specific order so the `delegate` is called when the attachments are really removed mimicking the regular remove method behavior as close as possible.

Reply:
Well the delegate isn't called when given attachment is removed, but when all are - the proposed solution does it in the moment when the specific attachment is removed.

It also doesn't create a copy of the array so it's more memory efficient, right?

Anyways I'll let other guys from MessageKit review it and we will get back to you here asap!

Reply:
AFAIK `let attachments = ...` should be unaffected by the subsequent `removeAll()` call. Also it is a `let`.

https://stackoverflow.com/a/24081412/1049134